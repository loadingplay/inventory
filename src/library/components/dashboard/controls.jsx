import React, { Component } from 'react';
import PropTypes from "prop-types";

import {
  LogoutButton,
  PickerClearButton,
  SearchInput,
  GenerateOrderButton,
  PrintButton
} from '../../components';


export class DashboardControls extends Component {

  static propTypes = {
    selectedCellarID: PropTypes.number,
    accessToken: PropTypes.string,
    onSearchTermChange: PropTypes.func,
    onPickerCleared: PropTypes.func
  }

  render = () => {
    return (
      <React.Fragment>
        <div className="col-6" >
          <SearchInput
            onSearch={this.props.onSearchTermChange}
            onTypeChange={this.props.onSearchTypeChange}
          ></SearchInput>
        </div>
        <div className="col-6 text-right">
          <PickerClearButton
            selected_cellar_id={this.props.selectedCellarID}
            onPickerClear={this.props.onPickerCleared}
          ></PickerClearButton>
          <PrintButton></PrintButton>
          <GenerateOrderButton
            selectedCellar={this.props.selectedCellarID}
            accessToken={this.props.accessToken}
            onOrderGenerated={this.props.onPickerCleared}
          ></GenerateOrderButton>
          <LogoutButton></LogoutButton>
        </div>
      </React.Fragment>
    );
  }
}
