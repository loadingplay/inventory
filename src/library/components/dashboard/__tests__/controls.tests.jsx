import React from 'react';
import { shallow } from "enzyme";

import { DashboardControls } from "../controls";

describe("DashboardControls", () => {
  test("it should render properly", () => {
    const wrapper = shallow(<DashboardControls />);
    expect(wrapper).toMatchSnapshot("detault");
  });
});
