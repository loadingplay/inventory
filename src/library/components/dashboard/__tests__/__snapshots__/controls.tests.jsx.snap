// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardControls it should render properly: detault 1`] = `
<Fragment>
  <div
    className="col-4"
  >
    <SearchInput />
  </div>
  <div
    className="col-8 text-right"
  >
    <PickerClearButton
      onPickerClear={[Function]}
    />
    <PrintButton />
    <GenerateOrderButton
      onOrderGenerated={[Function]}
    />
    <LogoutButton />
  </div>
</Fragment>
`;
